lib/w3c-trace-context.js

```diff
@@ -0,0 +1,252 @@
+'use strict'
+
+/* utility */
+
+// based on classic from: https://stackoverflow.com/a/38750895
+const filterObject = (unfiltered, allowed = []) => {
+  return Object.keys(unfiltered)
+    .filter(key => allowed.includes(key))
+    .reduce((obj, key) => {
+      obj[key] = unfiltered[key]
+      return obj
+    }, {})
+}
+
+/* validation */
+
+const validateXtrace = (xtrace) => {
+  // https://github.com/librato/trace/tree/master/docs/specs
+  const regExp = /\b2B[0-9A-F]{40}[0-9A-F]{16}0[0-1]{1}\b/
+  const matches = regExp.exec(xtrace)
+
+  return matches ? xtrace : ''
+}
+
+const validateTraceparent = (traceparent) => {
+  // https://www.w3.org/TR/trace-context/
+  const regExp = /\b00-[0-9a-f]{32}-[0-9a-f]{16}-0[0-1]{1}\b/
```

Comment:
`const regExp = /\b00-[0-9a-f]{32}-[0-9a-f]{16}-0[0-1]{1}\b/`
trace-flags can be `ff` in theory.
To make it more generic, we need to validate that as well.

Reply:
Agreed.
Now similar to [Ruby](https://github.com/appoptics/appoptics-apm-ruby/blob/8f570687a32609deb28b05c7605caf1cc647878f/lib/appoptics_apm/support/trace_parent.rb#L7) implementation.